tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks/ReadGitConfigFile.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Tasks;
+using Microsoft.Build.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class ReadGitConfigFile : Task
```

Comment:
Why wouldn't this be in the same task library as the VersionTools? It seems like we would need it for any other repo that does submodule auto-updates.

Reply:
In theory, the updaters can be configured from some other data source, but I agree--this way should be standard, and we should support it from BuildTools.

Reply:
Right now the implementation is also very specific to what you are after but the name is pretty general. It would probably be a good idea to make this a little more general purpose at some point.

Reply:
I started off with some more general output, but it was pretty unfriendly to try to get the info I wanted out of a more general config dump in MSBuild. I figured adding more outputs as they are needed might be a reasonable way to expand if necessary.